### PR TITLE
Refactor/avoid exposing pk to the outside of the system

### PR DESCRIPTION
**Pull request by YukinaMochizuki**
### Feature
- [add field should not updatable valid error type](https://github.com/charkchalk/backend/commit/58571a55797a04b15c92e3ef7240601583763fd4)
- [add generate slug helper method](https://github.com/charkchalk/backend/commit/2500268c96a64d994df30d8ea9b2cfd026ee5b48)
- [add put method to time range controller](https://github.com/charkchalk/backend/commit/e241d49ed3393bcd063d7e2f42cf7d835e4743a6)

### refactor
- [refactor: remove id field in tag response](https://github.com/charkchalk/backend/commit/473178c8294f8a423458f2f71b4130f12ee109e9)
- [refactor: remove id field in date range response](https://github.com/charkchalk/backend/commit/534f19ad1ea4b89ed0454ec171babf79e3c57044)
- [refactor: remove id field in date organization response](https://github.com/charkchalk/backend/commit/8f84f830e2e2d4e681fdc9f6e34f6950fda9dba1)

### CODE REVIEW

File: src/main/java/io/charkchalk/backend/exception/FieldNotValidType.java [hunk 1]
```diff
@@ -4,4 +4,5 @@ public enum FieldNotValidType {
     EntityNotFound,
     EntityAlreadyExists,
     SortPropertyNotFound,
+    FieldShouldNotUpdatable
```
Comment (moontai0724) [inline src/main/java/io/charkchalk/backend/exception/FieldNotValidType.java]: Maybe `FieldNotUpdatable` better?

File: src/main/java/io/charkchalk/backend/exception/FieldNotValidItem.java [hunk 2]
```diff
@@ -29,4 +29,9 @@ public static FieldNotValidItem sortPropertyNotFound(String property, String ent
         String message = String.format("No property %s found in entity %s", property, entity);
         return new FieldNotValidItem("sort", FieldNotValidType.SortPropertyNotFound, message);
     }
+
+    public static FieldNotValidItem entityFieldShouldNotUpdatable(String fieldName, String entity) {
```
Comment (moontai0724) [inline src/main/java/io/charkchalk/backend/exception/FieldNotValidItem.java]: Maybe `entityFieldNotUpdatable` better?

File: src/main/java/io/charkchalk/backend/json/JsonConverter.java [hunk 3]
```diff
@@ -32,4 +33,9 @@ public static List<FieldNotValidItem> checkPageableSortProperty(Pageable pageabl
 
         return fieldNotValidItems;
     }
+
+    public static String generateSlug(String name) {
+        String randomString = UUID.randomUUID().toString().replace("-","").substring(0,8);
+        return name.toLowerCase().replace(" ", "-") + "-" + randomString;
```
Comment (moontai0724) [inline src/main/java/io/charkchalk/backend/json/JsonConverter.java:42]: may not nessessary to add column name into slug

File: src/main/java/io/charkchalk/backend/json/JsonConverter.java [hunk 4]
```diff
@@ -32,4 +33,9 @@ public static List<FieldNotValidItem> checkPageableSortProperty(Pageable pageabl
 
         return fieldNotValidItems;
     }
+
+    public static String generateSlug(String name) {
+        String randomString = UUID.randomUUID().toString().replace("-","").substring(0,8);
```
Comment (moontai0724) [inline src/main/java/io/charkchalk/backend/json/JsonConverter.java]: I would prefer to save full UUID for further possibilities like duplication

**Review by moontai0724 — APPROVED** [review]

File: src/main/java/io/charkchalk/backend/json/JsonConverter.java (hunk 4 above)
Comment (moontai0724) [reply src/main/java/io/charkchalk/backend/json/JsonConverter.java]: btw, I mean to save full in database but be able to query by only first 8 digits

Comment (moontai0724) [reply src/main/java/io/charkchalk/backend/json/JsonConverter.java]: btw, I mean to save full in database but be able to query by only first n digits

Comment (YukinaMochizuki) [reply src/main/java/io/charkchalk/backend/json/JsonConverter.java]: I see, but wait for the first version of the specification to be completed before discussing

**Pull request merged**